src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/PerformingOrganisationNameAndAddress.java

```diff
@@ -0,0 +1,64 @@
+package uk.nhs.digital.nhsconnect.lab.results.model.edifact;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.Setter;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.message.EdifactValidationException;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.message.Split;
+
+/**
+ * Example:
+ *
+ * S01+01'
+ * NAD+SLA+++ST JAMES?'S UNIVERSITY HOSPITAL'
+ */
+@Getter
+@Setter
+@Builder
+@AllArgsConstructor
+public class PerformingOrganisationNameAndAddress extends Segment {
+
+    private static final String KEY = "NAD";
+    private static final String QUALIFIER = "SLA";
+    private static final String KEY_QUALIFIER = KEY + "+" + QUALIFIER;
+    private static final int PERFORMING_ORGANISATION_NAME_INDEX_IN_EDIFACT_STRING = 4;
+
+    private @NonNull String performingOrganisationName;
+
+    public static PerformingOrganisationNameAndAddress fromString(String edifactString) {
+        if (!edifactString.startsWith(PerformingOrganisationNameAndAddress.KEY_QUALIFIER)) {
+            throw new IllegalArgumentException(
+                "Can't create " + PerformingOrganisationNameAndAddress.class.getSimpleName() + " from " + edifactString
+            );
+        }
+
+        String[] keySplit = Split.byPlus(edifactString);
+        String performingOrganisationName = keySplit[PERFORMING_ORGANISATION_NAME_INDEX_IN_EDIFACT_STRING];
+
+        return new PerformingOrganisationNameAndAddress(performingOrganisationName);
+    }
+
+    @Override
+    public String getKey() {
+        return KEY;
+    }
+
+    @Override
+    public String getValue() {
+        return QUALIFIER + "+++" + performingOrganisationName;
+    }
+
+    @Override
+    protected void validateStateful() throws EdifactValidationException {
+
+    }
+
+    @Override
+    public void preValidate() throws EdifactValidationException {
+        if (performingOrganisationName.isEmpty()) {
```

Comment:
We could check if the string contains only whitespace:
```suggestion
        if (performingOrganisationName.isBlank()) {
```